control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go

```diff
@@ -71,9 +74,37 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 
 	// Setting the management state is required in order to create
 	// this object. We need to create this object before the cno starts
-	// because mutating many of the values (like vxlanport) is not premitted
+	// because mutating many of the values (like vxlanport) is not permitted
 	// after the cno reconciles this operator CR
 	if network.Spec.ManagementState == "" {
 		network.Spec.ManagementState = "Managed"
 	}
 }
+
+func DetectSuboptimalMTU(ctx context.Context, mgmtClient client.Client,
+	guestNetworkOperator *operatorv1.Network, hcp *hyperv1.HostedControlPlane) error {
+	const recommendedMinMTU = uint32(9000)
+	const ovnkOverhead = 200
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		hcp.Spec.Networking.NetworkType == hyperv1.OVNKubernetes &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig != nil &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU != nil &&
+		*guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU < recommendedMinMTU-ovnkOverhead {
+		meta.SetStatusCondition(&hcp.Status.Conditions, metav1.Condition{
+			Type:               string(hyperv1.SubOptimalMTUDetected),
+			Status:             metav1.ConditionTrue,
+			Reason:             hyperv1.KubeVirtSuboptimalMTUReason,
+			ObservedGeneration: hcp.Generation,
+			Message: fmt.Sprintf("A suboptimal MTU size of %d bytes has been detected. "+
+				"Due to performance, we recommend setting an MTU of %d bytes or greater for the "+
+				"infra cluster that hosts the KubeVirt VirtualMachines. When smaller MTUs are used, the cluster will "+
+				"still operate, but network performance will be degraded due to fragmentation of the double "+
+				"encapsulated ovn-k.",
```

Comment:
I'm concerned that displaying the exact MTU's discovered within the nested OVN podnet will lead to confusion.

while nested ovn is the case for 4.14, it is also not guaranteed moving forward that nested OVN will always be in use.  We will likely need to support secondary interfaces that my or may not be backed by OVN.

could we simplify this by saying that an MTU less that 9000 is detected and indicate that configuration could cause suboptimal network performance... so essentially what you have now, but not say what we detected on our end exactly, because I don't want users needing to do the overhead calculations to understand what all this means.

Reply:
done, disclosing the discovered MTU from guest's NetworkOperator is removed.